assets/js/base/hooks/use-container-queries.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { useResizeObserver } from 'wordpress-compose';
+
+/**
+ * Returns a resizeListener element and a class name based on its width.
+ * Class names are based on the smaller of the breakpoints:
+ * https://github.com/WordPress/gutenberg/tree/master/packages/viewport#usage
+ * Values are also based on those breakpoints minus ~80px which is aproximately
+ * the left + right margin in Storefront with a font-size of 16px.
+ * _Note: `useContainerQueries` will return an empty class name `` until after
+ * first render_
+ *
+ * @return {Array} An array of {Element} `resizeListener` and {string} `className`.
+ *
+ * @example
+ *
+ * ```js
+ * const App = () => {
+ * 	const [ resizeListener, containerQueryClassName ] = useContainerQueries();
+ *
+ * 	return (
+ * 		<div className={ containerQueryClassName }>
+ * 			{ resizeListener }
+ * 			Your content here
+ * 		</div>
+ * 	);
+ * };
+ * ```
+ */
+export const useContainerQueries = () => {
+	const [ resizeListener, { width } ] = useResizeObserver();
+
+	let className = '';
+	if ( width > 700 ) {
+		className = 'is-large';
+	} else if ( width > 520 ) {
+		className = 'is-medium';
+	} else if ( width > 400 ) {
```

Comment:
Longer term might be nice to have a way for client code to parameterise/customise the breakpoints. This is great though :)

Reply:
Yes, that's a great idea. For the _Cart_ and _Checkout_ block, this breakpoints were working fine, but if we see this hook being used in other blocks that require other breakpoints, we could let the consumer pass the breakpoints/classnames as a parameter.